easybuild/tools/filetools.py

```diff
@@ -1333,14 +1333,24 @@ def apply_patch(patch_file, dest, fn=None, copy=False, level=None, use_git_am=Fa
     return True
 
 
-def apply_regex_substitutions(paths, regex_subs, backup='.orig.eb'):
+def apply_regex_substitutions(paths, regex_subs, backup='.orig.eb', on_missing_match=None):
     """
     Apply specified list of regex substitutions.
 
     :param paths: list of paths to files to patch (or just a single filepath)
     :param regex_subs: list of substitutions to apply, specified as (<regexp pattern>, <replacement string>)
     :param backup: create backup of original file with specified suffix (no backup if value evaluates to False)
+    :param on_missing_match: Define what to do when no match was found in the file.
+                             Can be 'error' to raise an error, 'warn' to print a warning or 'ignore' to do nothing
+                             Defaults to value of --strict
```

Comment:
`None` should be mentioned there too, since that's the default value.
And perhaps the default should be `'strict'` (which corresponds to following `--strict`)

Reply:
I didn't want to mention `None` as that shouldn't be used. In a perfect world there would be no default value, I only added it for backwards compatibility. The caller should make a conscious decision what to do.
Hence also the use of `None` to represent "nothing passed".